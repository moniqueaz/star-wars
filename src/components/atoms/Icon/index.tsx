import * as S from './styles';
import { StyledIcon } from '@styled-icons/styled-icon';
import { RightArrow, User } from '@styled-icons/boxicons-regular';

type IconProps = {
  width?: string
  icon: string
}

type ListIconsProps = {
  [key: string]: StyledIcon
}

const listIcons = (): ListIconsProps => ( {
  arrow: RightArrow,
  user: User,
} );

const Icon = ( { width='4rem', icon } : IconProps) => {
  const Component = listIcons()[icon];

  if (!Component) {
    return null;
  }

  return (
    <S.Wrapper>
      <Component width={width}/>
    </S.Wrapper>
  );
};

export default Icon;
