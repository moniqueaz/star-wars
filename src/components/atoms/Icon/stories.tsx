import { Story, Meta } from '@storybook/react/types-6-0';
import Icon from '.';

export default {
  title: 'Atoms/Icon',
  component: Icon,
} as Meta;

export const Arrow: Story = args => {
  return <Icon icon="arrow" {...args}/>;
};

export const User: Story = args => {
  return <Icon icon="user" {...args}/>;
};
