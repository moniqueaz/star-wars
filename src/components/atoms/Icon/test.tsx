import { render } from '@testing-library/react';

import Icon from '.';

describe('<Icon />', () => {
  it('should render', () => {
    const { container } = render(<Icon icon="arrow"/>);
    expect(container.querySelector('svg')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  } );

  it('should render without icon', () => {
    const { container } = render(<Icon icon="none"/>);
    expect(container.querySelector('svg')).not.toBeInTheDocument();
  } );
} );
