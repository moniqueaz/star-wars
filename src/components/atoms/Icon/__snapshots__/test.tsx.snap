// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render 1`] = `
.c0 {
  width: auto;
  display: inline-block;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    class="c1"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    width="4rem"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
    />
  </svg>
</div>
`;
